### PR TITLE
refactor endpoints and services

### LINKED ISSUE #67: check over all end points
- [ ]  Delete unused end point 
- [ ]  Write Security filter with roles for endpoints

### PR DESCRIPTION
**Pull request by gerigol**
delete unnecessary endpoint, change nullchecking strategy, convert for cycles to streams

### CODE REVIEW

File: src/main/java/com/codecool/ecosampler/service/ProjectService.java [hunk 1]
```diff
@@ -29,30 +28,40 @@ public class ProjectService {
...
-        }
+        Project project = new Project(
+                UUID.randomUUID(),
+                newProject.name(),
+                newProject.description()
+        );
+        @NonNull List<UUID> userIDs = newProject.userIDs();
+        List<User> users = userService.getUsersByPublicId(userIDs);
+        project.addUsersToProject(users);
+
```
Comment (tuszmak) [inline src/main/java/com/codecool/ecosampler/service/ProjectService.java]: remove empty line

File: src/main/java/com/codecool/ecosampler/service/ProjectService.java [hunk 2]
```diff
@@ -63,33 +72,35 @@ public ProjectDTO updateProject(UUID publicId, ProjectDTO requestProject) {
...
 
     protected Project updateProjectWithRequest(ProjectDTO requestProject, Project project) {
-        if (Objects.nonNull(requestProject.name())) project.setName(requestProject.name());
+        @NonNull String projectName = requestProject.name();
+        project.setName(projectName);
 
-        if (Objects.nonNull(requestProject.description())) project.setDescription(requestProject.description());
+        @NonNull String projectDescription = requestProject.description();
+        project.setDescription(projectDescription);
         // TODO The rest
```
Comment (tuszmak) [inline src/main/java/com/codecool/ecosampler/service/ProjectService.java]: remove this todo, we don't implement anything else

File: src/main/java/com/codecool/ecosampler/service/QuestionService.java [hunk 3]
```diff
@@ -2,33 +2,31 @@
...
-        return questionRepository.findAll().stream()
-                .map(QuestionMapper::toDTO)
-                .collect(Collectors.toList());
-    }
-
     public QuestionDTO createQuestion(NewQuestion newQuestion) {
-        if(isQuestionExistByDescription(newQuestion.description())){
-            throw new BadRequestException("This question already exists: " + newQuestion.description());
+        if (isQuestionExistByDescription(newQuestion.description())) {
+            throw new BadRequestException(" Question already exists by description: " + newQuestion.description());
```
Comment (tuszmak) [inline src/main/java/com/codecool/ecosampler/service/QuestionService.java]: There's a space in the beginning of the error message.

**Review by tuszmak — CHANGES_REQUESTED** [review]

File: src/main/java/com/codecool/ecosampler/service/ProjectService.java [hunk 4]
```diff
@@ -63,33 +72,35 @@ public ProjectDTO updateProject(UUID publicId, ProjectDTO requestProject) {
...
-            project.removeUsersFromProject(removeUsers);
-        }
+        @NonNull List<UUID> addedUserIDs = modifyUsersOnProject.addUserIDs();
+        List<User> addUsers = userService.getUsersByPublicId(addedUserIDs);
+        project.addUsersToProject(addUsers);
+
+        @NonNull List<UUID> removedUserIDs = modifyUsersOnProject.removeUserIDs();
+        List<User> removeUsers = userService.getUsersByPublicId(removedUserIDs);
+        project.removeUsersFromProject(removeUsers);
+
```
Comment (MateWagner) [inline src/main/java/com/codecool/ecosampler/service/ProjectService.java]: We need to run add user if not null and same as to remove user

**Review by tuszmak — APPROVED** [review]

**Pull request merged**